pkg/execution/batch/redis.go

```diff
@@ -33,6 +37,43 @@ type redisBatchManager struct {
 	q redis_state.QueueManager
 }
 
+func (b redisBatchManager) batchKey(ctx context.Context, evt event.Event, fn inngest.Function) (string, error) {
+	if fn.EventBatch.Key == nil {
+		return fn.ID.String(), nil
+	}
+
+	out, _, err := expressions.Evaluate(ctx, *fn.EventBatch.Key, map[string]any{"event": evt.Map()})
+	if err != nil {
+		log.From(ctx).Error().Err(err).
+			Str("expression", *fn.EventBatch.Key).
+			Interface("event", evt.Map()).
+			Msg("error evaluating batch key expression")
+		return "<invalid>", nil
```

Comment:
if it's invalid, maybe just return it as the default `fn.ID.String()` instead?
hardcoding `"<invalid>"` here is no different from just using the function ID.

maybe also return an `error` just in case. I suspect it's not going to be used anytime soon, but hopefully we can surface this kind of invalid expression errors to users better.

Reply:
Update this to return an error and use the fn, just to clarify, should we pass this up already or silently ignore the error? I guess it does make sense to warn early and condition users to use valid expressions returning strings, what do you think?

Reply:
We don't have a way to pass this error back up to the user atm.
It's the same for all expression related evaluation, so we should bubble the evaluation error up somehow so users are aware.

But regardless we don't want to `fail` the execution because there's an invalid expression since there are no ways to recover from that and runs are essentially lost at that point.